sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,475 @@
+macro `SYCL_EXT_ONEAPI_COMPLEX_MARRAY` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== Marray Complex Class Specialization
+
+The user interface of the `marray<sycl::ext::oneapi::complex, {N}>`
+class is changed compared to the SYCL-2020 generic `marray` interface.
+It adds new constructors, `real()`, and `imag()`. This changes the
+`marray` interface when specialized for the `sycl::ext::oneapi::complex`
+type.
+
+The `marray` complex specialization is trivially copyable, and the type
+trait `is_device_copyable` should resolve to `std::true_type`.
+
+The `marray` definition used within this proposal assumes that any
+operator the `marray` class defines is only implemented if the marray's
+value type also implements the operator. So, for example,
+`marray<float, N>` does not implement the modulus operator as float does
+not support it. The specialized marray class is shown below.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+    // Specialization of exiting `marray` class for `sycl::ext::oneapi::complex`
+    template <typename T, std::size_t NumElements>
+    class marray <sycl::ext::oneapi::complex<T>, NumElements> {
+        private:
+            using DataT = sycl::ext::oneapi::complex<T>;
+        public:
+            using value_type = DataT;
+            using reference = DataT&;
+            using const_reference = const DataT&;
+            using iterator = DataT*;
+            using const_iterator = const DataT*;
+
+            marray();
+
+            explicit constexpr marray(const DataT &arg);
+
+            template <typename... ArgTN>
+            constexpr marray(const ArgTN&... args);
+
+            constexpr marray(const marray<DataT, NumElements> &rhs);
+            constexpr marray(marray<DataT, NumElements> &&rhs);
+
+            // Additional constructors
+            constexpr marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag);
+
+            constexpr marray(const marray<T, NumElements> &real, T imag);
+
+            constexpr marray(T real, const marray<T, NumElements> &imag);
+
+            template <std::size_t... I>
+            constexpr marray(const marray<T, NumElements> &real, const marray<T, NumElements> &imag, std::integer_sequence<std::size_t, I...> int_seq);
+
+            template <std::size_t... I>
+            constexpr marray(const marray &cmplx, std::integer_sequence<std::size_t, I...> int_seq);
+
+            // Available only when: NumElements == 1
+            operator DataT() const;
+
+            static constexpr std::size_t size() noexcept;
+
+            // real and imag
```

Comment:
@TApplencourt The problem that we were talking about with the ambiguous getter and setters, as well as the free functions, has already been discussed here.